src/Features/Core/Diagnostics/AbstractHostDiagnosticUpdateSource.cs

```diff
@@ -44,14 +44,23 @@ protected void RaiseDiagnosticsUpdated(DiagnosticsUpdatedArgs args)
             }
         }
 
-        internal void ReportAnalyzerDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, Workspace workspace, ProjectId projectId)
+        internal void ReportAnalyzerDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, Workspace workspace, ProjectId projectOpt)
         {
             if (workspace != this.Workspace)
             {
                 return;
             }
 
-            var project = workspace.CurrentSolution.GetProject(projectId);
+            // check whether we are reporting project specific diagnostic or workspace wide diagnostic
+            var project = projectOpt != null ? workspace.CurrentSolution.GetProject(projectOpt) : null;
+            
+            // check whether project the diagnostic belong to still exist
+            if (projectOpt != null && project == null)
```

Comment:
@shyamnamboodiripad yes that is intentional. Null projectId argument means the analyzer exception diagnostic is during basic analyzer initialization (say from SupportedDiagnostics invoked on VSIX based analyzer, which happens in no particular project context). Such diagnostics are analyzer specific, and not specific to any project and hence reported as "workspace" diagnostics later in this method.